sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/QueueAysncAPITests.groovy

```diff
@@ -0,0 +1,548 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.spock
+
+import com.azure.storage.queue.models.AccessPolicy
+import com.azure.storage.queue.models.SignedIdentifier
+import com.azure.storage.queue.models.StorageErrorCode
+import org.junit.Ignore
+import reactor.test.StepVerifier
+import spock.lang.Unroll
+
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+
+import static org.junit.Assert.assertEquals
+
+class QueueAysncAPITests extends APISpec {
+    def queueAsyncClient
+
+    static def testMetadata = Collections.singletonMap("metadata", "value")
+    static def createMetadata = Collections.singletonMap("metadata1", "value")
+
+    def setup() {
+        primaryQueueServiceAsyncClient = queueServiceBuilderHelper(interceptorManager).buildAsyncClient()
+        queueAsyncClient = primaryQueueServiceAsyncClient.getQueueAsyncClient(testResourceName.randomName("queue", 16))
+    }
+
+    def "Create queue with shared key from queue async client"() {
+        when:
```

Comment:
Could this entire test just be an expect statement?

Reply:
Done.